encoding/flag/flag.go

```diff
@@ -0,0 +1,170 @@
+package flag
```

Comment:
This is an import conflict, should we change the package to flg or something similar?